### ISSUE TITLE
[BDD] Refine k-psp-003

### ISSUE THREAD

**Issue by eddie-knight** [citihub/probr#338]
The following changes have been proposed by @paulm-jones. Note that this assumes that the `Given...` has been extracted to the probe's background.

## Old scenario
```
    Scenario Outline: Prevent a deployment from running with the hostIPC flag.
        And some system exists to prevent a Kubernetes deployment from running using the hostIPC in an existing Kubernetes cluster
        When a Kubernetes deployment is applied to an existing Kubernetes cluster
        And hostIPC request is marked "<hostIPC access is requested>" for the Kubernetes deployment
        Then the operation will "<RESULT>" with an error "<ERRORMESSAGE>"
        But I should not be able to perform a command that provides access to the host IPC namespace
        And I should be able to perform an allowed command

        Examples:
            | hostIPC access is requested | RESULT   | ERRORMESSAGE                        |
            | True                        | Fail     | Containers with hostIPC access can't run |
            | False                       | Succeed  | No error would show                      |
            | Not defined                 | Succeed  | No error would show                      |
```

## New Scenario
```
  Scenario Outline: Prevent a deployment from running with the hostIPC flag.

  HostIPC controls whether a Pod's containers can share the host IPC namespace, allowing container processes to
  communicate with other processes on the host.

  See https://kubernetes.io/docs/concepts/policy/pod-security-policy/#host-namespaces.

    And I request creation of a Pod with shared host IPC namespace "<hostIPC access is requested>"
    Then the operation will "<RESULT>" with an error "<ERRORMESSAGE>"
    And I should be able to perform an allowed command
    But I should not be able to perform a command that provides access to the host IPC namespace

    Examples:
      | hostIPC access is requested | RESULT  | ERRORMESSAGE                             |
      | True                        | Fail    | Containers with hostIPC access can't run |
      | False                       | Succeed | No error would show                      |
      | Not defined                 | Succeed | No error would show                      |
```

**Comment by iantivey** [comment]
Recommended approach to simplifying this scenario:

```
Scenario: Prevent a deployment from running with privileged access

  HostIPC controls whether a Pod's containers can share the host IPC namespace, allowing container processes to
  communicate with other processes on the host.

  See https://kubernetes.io/docs/concepts/policy/pod-security-policy/#host-namespaces.

   And creation of a Pod explicitly specifying access to the shared host IPC namespace = false is successful
   When a user requests creation of the same Pod but requesting access to the shared host IPC namespace = true
   Then the Pod creation is unsuccessful due to restrictions in requesting access to the shared host IPC namespace

Scenario: Prevent execution of commands that allow access to the shared host IPC namespace access by default

  By default Pods that don't specify whether Host IPC namespace mode is set should not be able to access the shared host IPC namespace.

  See https://kubernetes.io/docs/concepts/policy/pod-security-policy/#privileged

    And creation of a Pod without specifying a value for shared host IPC namespace is successful
    When a user attempts to execute a command inside the Pod that provides access to the shared host IPC namespace
    Then the command execution is unsuccessful due to shared host IPC namespace access restrictions

Scenario: Prevent execution of commands that allow access to the shared host IPC namespace

  Pods that specifically request shared Host IPC namespace access be set to false should not be able to access the shared host IPC namespace

    And creation of a Pod explicitly specifying shared host IPC namespace = false is successful
    When a user attempts to execute a command inside the Pod that provides access to the shared host IPC namespace
    Then the command execution is unsuccessful due to shared host IPC namespace access restrictions
```

**Comment by eddie-knight** [comment]
closed by #374
